src/server/active_tran_server.cpp

```diff
@@ -186,13 +183,19 @@ active_tran_server::connection_handler::get_saved_lsa () const
 }
 
 void
-active_tran_server::connection_handler::remove_prior_sender_sink ()
+active_tran_server::connection_handler::on_connecting ()
```

Comment:
If the job processing in this function is going to heavy later by someone then, the prior_sender could be blocked some period time by m_state_mtx mutex(And other connection handler... ATS -> PS(es)). So, how about checking m_state first before getting the shared mutex m_state_mtx in the tran_server::connection_handler::push_request().

Reply:
I understand the prior_sender can get blocked since m_state_mtx is locked exclusively during on_creating() and prior_sender iterates sink_hooks sequentially. However, I don't understand the solution. We can't trust the value of m_state without m_state_mtx. Do I get it right?

Anyway, the problem is valid and we should avoid that situation. I think we can avoid that with releasing m_state_mtx before on_creating() and re-acquiring the mutex after the function. What do you think?

Reply:
I need to time to check your suggestion. I will come back soon hear. My suggestion is the following:
```
if (m_state != CONNECTED)
{
        // give-up this time, try next call
}

acquire m_state_mtx mutex

// check again
if (m_state != CONNECTED)
{
        // give-up
}
```

Reply:
@wrlawodms Before start review again ..

Is there any reason the prior_sender has to know(check) the state of conection hander? Please remind me. I expect that prior_sender doesn't check the state of connection handler like CONNECTING, CONNECTED. Only CONNECTED connection is added to the prior_sender and prior_sender just call the function(push_request) without checking the state.

Reply:
`push_request` checks the state and only sends requests when "CONNECTED". It acquires the mutex to check it. By the way, now it has a problem. The prior_sender has to send the prior list even during CONNECTING because log records have to be sent during the catch-up and the state of the PS is CONNECTING during the job, I realized it just yesterday. It will be fixed with http://jira.cubrid.org/browse/LETS-768.

All in all, the prior send
- the prior_sender will use another function as the prior sink hook, not push_request(). it will acquire the m_conn_mtx only, but not m_state_mtx.
- In this case, on_creating with m_state_mtx doesn't matter for the problem you mentioned. But, I think I should release m_state_mtx before on_connecting() and on_disconnecting(). It will reduce the lock scope.

Reply:
> push_request checks the state and only sends requests when "CONNECTED". It acquires the mutex to check it.
I know it aready. I told that connection is added to prior_sender after changing the state to the CONNECTED. It mean that prior_sender only have CONNECTED state connections. And prior_sender doesn't consider the state of the connection_hander.

Reply:
I might miss something, but the prior sink hook is added before the state is set to CONNECTED in `connect()`. It has to be done for the catch-up or some preliminary communication before being set to CONNECTED.

Reply:
OK. Let's see in the LETS-768 :D